src/Common/TagStoreInterface.php

```diff
@@ -4,15 +4,66 @@
 
 interface TagStoreInterface
 {
-    public function tagObject($objectId, array $tags);
+    /**
+     * Associate the given tags to cache item identified by $itemId.
+     *
+     * Example: associate a HTTP cache identifier (e.g. a content-digest in the
+     *          case of the Symfony HTTP cache) with a set of tags.
+     *
+     * @param integer $itemId
+     * @param string[] $tags
+     */
+    public function tagItemId($itemId, array $tags);
 
-    public function untagObject($objectId, array $tags);
+    /**
+     * Unassociate the given tags from the cache item identified by $itemId.
+     *
+     * NOTE: This is difficult for an so-called "cache based" implementation, is it necessary?
+     *
+     * @param integer $itemId
+     * @param string[] $tags
+     */
+    public function untagItemId($itemId, array $tags);
```

Comment:
> The bare minimum of methods I'd consider necessary for an HTTP cache is probably

Agree. For the SF HTTP Cache `4` is not essential, but as you say it may be more important in other caches, and it helps with house-keeping.

> As for untagItemId and getTagsForItemIds, they're mostly there for completeness, but could potentially be useful for debugging and introspection.

I would personally tend towards droping them from the API, if it becomes apparent that we need them later on then we can easily add them.